src/_ert/events.py

```diff
@@ -0,0 +1,202 @@
+import sys
+from datetime import datetime
+from typing import Any, Dict, Literal, Union
+
+if sys.version_info < (3, 9):
+    from typing_extensions import Annotated
+else:
+    from typing import Annotated
+
+from pydantic import BaseModel, Field, TypeAdapter
+
+_FORWARD_MODEL_START = Literal["com.equinor.ert.forward_model_job.start"]
+_FORWARD_MODEL_RUNNING = Literal["com.equinor.ert.forward_model_job.running"]
+_FORWARD_MODEL_SUCCESS = Literal["com.equinor.ert.forward_model_job.success"]
+_FORWARD_MODEL_FAILURE = Literal["com.equinor.ert.forward_model_job.failure"]
+_FORWARD_MODEL_CHECKSUM = Literal["com.equinor.ert.forward_model_job.checksum"]
+
+_REALIZATION_FAILURE = Literal["com.equinor.ert.realization.failure"]
+_REALIZATION_PENDING = Literal["com.equinor.ert.realization.pending"]
+_REALIZATION_RUNNING = Literal["com.equinor.ert.realization.running"]
+_REALIZATION_SUCCESS = Literal["com.equinor.ert.realization.success"]
+_REALIZATION_UNKNOWN = Literal["com.equinor.ert.realization.unknown"]
+_REALIZATION_WAITING = Literal["com.equinor.ert.realization.waiting"]
+_REALIZATION_TIMEOUT = Literal["com.equinor.ert.realization.timeout"]
+
+_ENSEMBLE_STARTED = Literal["com.equinor.ert.ensemble.started"]
+_ENSEMBLE_SUCCEEDED = Literal["com.equinor.ert.ensemble.succeeded"]
+_ENSEMBLE_CANCELLED = Literal["com.equinor.ert.ensemble.cancelled"]
+_ENSEMBLE_FAILED = Literal["com.equinor.ert.ensemble.failed"]
+
+_EE_SNAPSHOT = Literal["com.equinor.ert.ee.snapshot"]
+_EE_SNAPSHOT_UPDATE = Literal["com.equinor.ert.ee.snapshot_update"]
+_EE_TERMINATED = Literal["com.equinor.ert.ee.terminated"]
+_EE_USER_CANCEL = Literal["com.equinor.ert.ee.user_cancel"]
+_EE_USER_DONE = Literal["com.equinor.ert.ee.user_done"]
+
+
+class BaseEvent(BaseModel):
+    time: datetime = Field(default_factory=datetime.now)
+
+
+class ForwardModelBaseEvent(BaseEvent):
+    ensemble: Union[str, None] = None
+    real: str
+    fm_step: str
+
+
+class ForwardModelStart(ForwardModelBaseEvent):
+    event_type: _FORWARD_MODEL_START = "com.equinor.ert.forward_model_job.start"
+    std_out: Union[str, None] = None
+    std_err: Union[str, None] = None
+
+
+class ForwardModelRunning(ForwardModelBaseEvent):
+    event_type: _FORWARD_MODEL_RUNNING = "com.equinor.ert.forward_model_job.running"
+    max_memory_usage: Union[int, None] = None
+    current_memory_usage: Union[int, None] = None
+
+
+class ForwardModelSuccess(ForwardModelBaseEvent):
```

Comment:
`ForwardModelStepSuccess`, but for all I guess. @berland

Reply:
Agree, should include "step" wherever relevant.

Reply:
added step to class names